### DISCUSSION TITLE
ONDC Mobility engagement framework

### DISCUSSION THREAD

**Discussion by PrakashRydo** [ONDC-Official/mobility-specification#6] [General]
In a ride hailing platform the ride details collection, route estimation and distance calculation for fare estimation is a critical function and it has direct bearing on the cost and efficiency of the platform. In an open network scenario we have this question
● Who should do the route and distance calculation? BAP or BPP?
This is significant because map costs and service choices are involved in this process. I have shared my thoughts in the attached deck on the framework engagement in the following slides and welcome inputs from all.
[ONDC ride discovery framework.pdf](https://github.com/ONDC-Official/mobility-specification/files/11755795/ONDC.ride.discovery.framework.pdf)

**Comment by NirmalR-protean** [comment]
Hi @PrakashRydo, in what form would the route be transmitted to the BAP in option 1? 

Usually in apps I have used, from an end user only the start and end location is taken. The route the driver takes would be shown but it cannot be edited in any way by the user (apart from adding a stop inbetween). Even while the ride is in progress the driver may take another route based on real time parameters like traffic.  So I think it makes sense that BAP will transmit just the start and end location from the user and the BPP can respond with available services that can achieve the same.

**Reply by shobhanaquickride** [reply]
I agree; the recommended route and hence the distance should be determined by the BPP and the cost would depend on the same.
